x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/custom_timeline_data_grid_body.tsx

```diff
@@ -122,34 +160,48 @@ const CustomDataGridSingleRow = memo(function CustomDataGridSingleRow(
   );
   const eventTypeRowClassName = useGetEventTypeRowClassName(rowData.ecs);
 
+  const isRowIntersecting =
+    intersectionEntry.isIntersecting && intersectionEntry.intersectionRatio > 0;
+
+  const isRowLoading = IS_ROW_RENDERER_LAZY_LOADING_ENABLED && !isRowIntersecting;
+
   return (
     <CustomGridRow
-      className={`${rowIndex % 2 === 0 ? 'euiDataGridRow--striped' : ''}`}
+      className={`${rowIndex % 2 === 0 && !isRowLoading ? 'euiDataGridRow--striped' : ''}`}
       key={rowIndex}
+      ref={intersectionRef}
     >
-      <CustomGridRowCellWrapper className={eventTypeRowClassName}>
-        {visibleColumns.map((column, colIndex) => {
-          // Skip the expanded row cell - we'll render it manually outside of the flex wrapper
-          if (column.id !== TIMELINE_EVENT_DETAIL_ROW_ID) {
-            return (
-              <Cell
-                style={cellCustomStyle}
-                colIndex={colIndex}
-                visibleRowIndex={rowIndex}
-                key={`${rowIndex},${colIndex}`}
-              />
-            );
-          }
-          return null;
-        })}
-      </CustomGridRowCellWrapper>
-      {/* Timeline Expanded Row */}
-      {canShowRowRenderer ? (
-        <Cell
-          colIndex={visibleColumns.length - 1} // If the row is being shown, it should always be the last index
-          visibleRowIndex={rowIndex}
-        />
-      ) : null}
+      {isRowLoading ? (
+        <CustomLazyRowPlaceholder />
```

Comment:
Could we use the skeleton here? https://eui.elastic.co/#/display/skeleton

Reply:
I used skeleton first and then rejected it because showing rows loading when scrolling makes the UI feel sluggish. If you want we can have a meeting and I can demo both.